ui/src/app/cohort-search/attributes-page-v2/attributes-page-v2.component.tsx

```diff
@@ -0,0 +1,610 @@
+            modifiers: []
+          }],
+          temporal: false
+        }],
+        dataFilters: []
+      };
+      cohortBuilderApi().countParticipants(cdrVersionId, request).then(response => {
+        this.setState({count: response, calculating: false});
+      }, () => {
+        this.setState({calculating: false, countError: true});
+      });
+    }
+
+    addParameterToSearchItem() {
+      const {close} = this.props;
+      let {criteria} = this.props;
+      const param = this.paramWithAttributes;
+      // TODO remove condition to only track PM criteria for 'Phase 2' of CB Google Analytics
+      if (this.isPhysicalMeasurement) {
+        this.trackEvent(param.subtype, 'Add');
+      }
+      criteria = criteria.filter(crit => crit.parameterId !== param.parameterId);
+      currentCohortCriteriaStore.next([...criteria, param]);
+      close();
+    }
+
+    trackEvent(subtype: string, eventType: string) {
+      triggerEvent(
+        'Cohort Builder Search',
+        'Click',
+        `Physical Measurements - ${subTypeToTitle(subtype)} - ${eventType}`
+      );
+    }
+
+    get hasUnits() {
+      const {node: {subtype}} = this.props;
+      return typeof PM_UNITS[subtype] !== 'undefined';
+    }
+
+    get isMeasurement() {
+      const {node: {domainId}} = this.props;
+      return domainId === DomainType.MEASUREMENT;
+    }
+
+    get isPhysicalMeasurement() {
+      const {node: {domainId}} = this.props;
+      return domainId === DomainType.PHYSICALMEASUREMENT;
+    }
+
+    get isSurvey() {
+      const {node: {domainId}} = this.props;
+      return domainId === DomainType.SURVEY;
+    }
+
+    get isBloodPressure() {
+      const {node: {subtype}} = this.props;
+      return subtype === CriteriaSubType.BP;
+    }
+
+    get hasRange() {
+      return this.isMeasurement || this.isSurvey;
+    }
+
+    render() {
+      const {close, node: {domainId, name, parentId, subtype}} = this.props;
+      const {calculating, count, countError, form, loading, options} = this.state;
+      const {formErrors, formValid} = this.validateForm();
+      const disableAdd = calculating || !formValid;
+      const disableCalculate = disableAdd || form.exists || form.num.every(attr => attr.operator === 'ANY');
+      return (loading ?
+        <SpinnerOverlay/> :
+        <div style={{marginTop: '0.5rem'}}>
+          <h3 style={{fontWeight: 600, margin: '0 0 0.5rem', textTransform: 'capitalize'}}>
+            {domainId === DomainType.PHYSICALMEASUREMENT.toString() ? name : domainId.toString().toLowerCase()} Detail
+          </h3>
+          {countError && <div style={styles.error}>
+            <ClrIcon style={{margin: '0 0.5rem 0 0.25rem'}} className='is-solid'
+              shape='exclamation-triangle' size='22'/>
+            Sorry, the request cannot be completed.
+          </div>}
+          {!!formErrors.size && <div style={styles.errors}>
+            {Array.from(formErrors).map((err, e) => <div key={e} style={styles.errorItem}>
```

Comment:
Question: Can we use validate in this form?

Reply:
I know I tried using it here at some point, possibly when I converted to React. Can't remember for sure, but I want to say it was easier to get the error messaging and styles how we wanted them just writing custom validation. I'll have to take another look at it though.

Reply:
I think the main issue is that the form is dynamic, so depending on the type of attribute and the operator selected, there could be 1-4 values that need to be passed in to the `validate()` function.

Also, we need to validate values differently for different domains. Physical measurements values just need to be positive but attributes for Measurements have a specific range that can include negative numbers.

There's probably a way to do it, but seems like a decent amount work and I would prefer to handle it separately instead of including here. I can make a ticket for that.